### PR TITLE
ENG-5813 - TestFlight Actions

### PR DESCRIPTION
**Pull request by kevinsites**
This is a major clean up of the old actions to follow a more consistent release pattern we are moving towards. If you have any questions please let me know!

### CODE REVIEW

**Review by kishisaka-nid — DISMISSED** [review]
pulled, tested, run unit test, all 163 passed. LGTM. 🚢

**Review by priya-xavier — APPROVED** [review]
Nice workflow!

File: .github/workflows/ios-sandbox-deployment-dev.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+          path: Pods
+          key: ${{ runner.os }}-pods-${{ hashFiles('**/Podfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-pods-
+
+      - name: CocoaPod Install
+        run: pod install
+
+      - name: Save Updated Pods
+        id: cache-primes-save
```
Comment (priya-xavier) [inline .github/workflows/ios-sandbox-deployment-dev.yml:47]: id name is confusing- change primes to pods?

Comment (kevinsites) [reply .github/workflows/ios-sandbox-deployment-dev.yml:47]: yes good catch, I copied from their docs and didn't change that part 😅

File: .github/workflows/ios-sandbox-deployment-dev.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            -d '{"event_type":"on-demand-testflight-dev","client_payload":{"version":"5.ios-${{env.VERSION}}", "message": "${{ github.event.head_commit.message }}"}}'
+
+      - name: Run ReactNative Sandbox Dev Deployment
+        run: |
+          curl \
+              -X POST \
+              -H "Accept: application/vnd.github.v3+json" \
+              -H "Authorization: token ${{ secrets.GPR_API_KEY }}" \
+              https://api.github.com/repos/Neuro-ID/neuroid-reactnative-sdk/dispatches \
+              -d '{"event_type":"on-demand-testflight-dev","client_payload":{"version":"5.ios-${{env.VERSION}}", "message": "${{ github.event.commits[0].message }}"}}'
```
Comment (af-neuro) [inline .github/workflows/ios-sandbox-deployment-dev.yml:95]: Just FYI - message reference looks different than the rest (although they probably evaluate to the same value).

**Review by af-neuro — APPROVED** [review]

**Review by af-neuro — APPROVED** [review]

**Review by priya-xavier — APPROVED** [review]

**Pull request merged**